Python/compile.c

```diff
@@ -2567,6 +2570,42 @@ compiler_check_debug_args(struct compiler *c, arguments_ty args)
     return 1;
 }
 
+static inline int
+insert_instruction(basicblock *block, int pos, struct instr *instr) {
+    if (basicblock_next_instr(block) < 0) {
+        return -1;
+    }
+    for (int i = block->b_iused - 1; i > pos; i--) {
+        block->b_instr[i] = block->b_instr[i-1];
+    }
+    block->b_instr[pos] = *instr;
+    return 0;
+}
+
+static int
+wrap_in_stopiteration_handler(struct compiler *c)
+{
```

Comment:
I wonder if this is not wrapping too much, like the instructions that deal with the function's arguments and annotations?

Reply:
Annotations are executed in the enclosing scope, as are defaults. So there is no problem there.
`MAKE_CELL` and `RETURN_GENERATOR` instructions are inserted in the back-end so happen after this.